build.gradle.kts

```diff
@@ -1,105 +1,67 @@
-    `maven-publish`
-    id("org.ajoberstar.grgit") version "4.1.1"
+    id("java-library")
+    id("maven-publish")
     id("com.github.johnrengelman.shadow") version "7.1.2"
 }
 
-logger.lifecycle("""
-*******************************************
- You are building LWCX!
- If you encounter trouble:
- 1) Try running 'build' in a separate Gradle run
- 2) Use gradlew and not gradle
- 3) If you still need help, ask on Discord! https://discord.gg/PHpuzZS
-
- Output files will be in build/libs
-*******************************************
-""")
-
-if (!project.hasProperty("gitCommitHash")) {
-    apply(plugin = "org.ajoberstar.grgit")
-    ext["gitCommitHash"] = try {
-        Grgit.open(mapOf("currentDir" to project.rootDir))?.head()?.abbreviatedId
-    } catch (e: Exception) {
-        logger.warn("Error getting commit hash", e)
-
-        "no.git.id"
+java {
+    toolchain {
+        languageVersion.set(JavaLanguageVersion.of(8))
     }
+    withSourcesJar()
 }
 
 repositories {
     mavenCentral()
-
-    maven { url = uri("https://repo.papermc.io/repository/maven-public/") }
-    maven { url = uri("https://maven.enginehub.org/repo/") }
-    maven { url = uri("https://repo.glaremasters.me/repository/public/") }
-    maven { url = uri("https://jitpack.io") }
+    maven("https://hub.spigotmc.org/nexus/content/repositories/snapshots/")
+    maven("https://maven.enginehub.org/repo/")
+    maven("https://repo.glaremasters.me/repository/towny/")
+    maven("https://ci.ender.zone/plugin/repository/everything/")
+    maven("https://repo.glaremasters.me/repository/public/")
+    maven("https://jitpack.io")
 }
 
 configurations {
     compileClasspath.get().extendsFrom(create("shadeOnly"))
 }
 
 dependencies {
-    compileOnly("org.jetbrains:annotations:23.0.0")
-
-    compileOnly("org.spigotmc:spigot-api:1.19.2-R0.1-SNAPSHOT")
-    compileOnly("com.googlecode.json-simple:json-simple:1.1.1")
-    compileOnly("commons-lang:commons-lang:2.6")
-
-    compileOnly("com.sk89q.worldedit:worldedit-bukkit:7.2.12")
-    compileOnly("com.sk89q.worldguard:worldguard-bukkit:7.0.7")
-    compileOnly("com.github.TownyAdvanced:Towny:0.98.3.7")
-    compileOnly("com.massivecraft:Factions:1.6.9.5-U0.4.9") { isTransitive = false }
-    compileOnly("com.github.milkbowl:VaultAPI:1.7.1")
-    "shadeOnly"("org.bstats:bstats-bukkit:3.0.0")
-
-}
-
-tasks.named<Copy>("processResources") {
-    val internalVersion = "$version-${rootProject.ext["gitCommitHash"]}"
-    inputs.property("internalVersion", internalVersion)
-
-    filesMatching("plugin.yml") {
-        expand("internalVersion" to internalVersion)
+    compileOnly(group = "org.spigotmc", name = "spigot-api", version = "1.19.2-R0.1-SNAPSHOT")
+    compileOnly(group = "com.sk89q.worldedit", name = "worldedit-core", version = "7.1.0")
+    compileOnly(group = "com.sk89q.worldguard", name = "worldguard-bukkit", version = "7.0.0")
+    compileOnly(group = "com.palmergames.bukkit.towny", name = "towny", version = "0.98.2.0")
```

Comment:
I updated the dependency versions, not sure why you don't want to update them. There might be some deprecated things in the old version. You might miss those deprecations with outdated dependencies.

Reply:
There were breaking changes here, which were causing builds to fail. Let's keep these on safe versions for now to separate out concerns, we can update them in a follow-up PR.